s3/s3_transport/include/s3_transport.hpp

```diff
@@ -1067,58 +1067,90 @@ namespace irods::experimental::io::s3_transport
+                // Calculate the number of parts.
+                unsigned int number_of_parts = config_.number_of_cache_transfer_threads;
+                if (cache_file_size > number_of_parts * preferred_part_size) {
+                    number_of_parts
+                      = cache_file_size % preferred_part_size == 0
+                      ? cache_file_size / preferred_part_size
+                      : cache_file_size / preferred_part_size + 1;
+                }
 
-            if (config_.multipart_enabled && number_of_parts > 1) {
+                if (number_of_parts > 10000) {
+                    logger::error("Multiple timeouts resulted in a part size that is too small to upload the {} size file.", cache_file_size);
+                    return_value = error_codes::UPLOAD_FILE_ERROR;
+                    // bit redundant here
+                    break_out = true;
+                    break;
+                }
 
-                initiate_multipart_upload();
+                if (config_.multipart_enabled && number_of_parts > 1) {
 
-                unsigned int part_number = 1;
+                    initiate_multipart_upload();
 
-                std::int64_t part_size_all_but_last_part = cache_file_size / number_of_parts;
+                    unsigned int part_number = 1;
 
-                while (part_number <= number_of_parts) {
+                    std::int64_t part_size_all_but_last_part = cache_file_size / number_of_parts;
 
-                    irods::thread_pool cache_flush_threads{static_cast<int>(config_.number_of_cache_transfer_threads)};
+                    while (part_number <= number_of_parts) {
 
-                    // run number_of_cache_transfer_threads simultaneously
-                    for (unsigned int i = 0; i < config_.number_of_cache_transfer_threads; ++i) {
+                        irods::thread_pool cache_flush_threads{static_cast<int>(config_.number_of_cache_transfer_threads)};
 
-                        if (part_number > number_of_parts) {
-                            break;
-                        }
+                        // run number_of_cache_transfer_threads simultaneously
+                        for (unsigned int i = 0; i < config_.number_of_cache_transfer_threads; ++i) {
+
+                            if (part_number > number_of_parts) {
+                                break;
+                            }
 
-                        std::int64_t part_size = part_size_all_but_last_part;
+                            std::int64_t part_size = part_size_all_but_last_part;
 
-                        // give extra bytes to last part
-                        if (part_number == number_of_parts) {
-                            part_size += cache_file_size % number_of_parts;
-                        }
+                            // give extra bytes to last part
+                            if (part_number == number_of_parts) {
+                                part_size += cache_file_size % number_of_parts;
```

Comment:
Consider wrapping the expression on the right of `+=` in parens for readability and correctness.

Reply:
Done but I would disagree with the correctness comment as % has higher precedence than +=.